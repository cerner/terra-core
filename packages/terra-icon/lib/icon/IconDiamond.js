'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _IconBase = require('../IconBase');

var _IconBase2 = _interopRequireDefault(_IconBase);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SvgIcon = function SvgIcon(customProps) {
  var attributes = _extends({}, customProps);
  attributes.className = (0, _classnames2.default)("is-static", attributes.className);

  return _react2.default.createElement(
    _IconBase2.default,
    attributes,
    _react2.default.createElement('path', { fill: '#78288C', d: 'M24.8 47.7l22.9-22.9c.4-.4.4-1.1 0-1.6L24.8.3c-.4-.4-1.1-.4-1.6 0L.3 23.2c-.4.4-.4 1.1 0 1.6l22.9 22.9c.4.4 1.2.4 1.6 0z' })
  );
};

SvgIcon.displayName = "IconDiamond";
SvgIcon.defaultProps = { "className": "is-static", "viewBox": "0 0 48 48", "xmlns": "http://www.w3.org/2000/svg" };

exports.default = SvgIcon;