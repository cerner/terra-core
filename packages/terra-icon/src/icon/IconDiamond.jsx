/* eslint-disable */
import React from 'react';
import classNames from 'classnames/bind';
import IconBase from '../IconBase';
import styles from '../IconDiamond.module.scss';

const cx = classNames.bind(styles);

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);
  const iconClassNames = cx([
    'IconDiamond',
    customProps.className,
  ]);

  return (
    <IconBase className={iconClassNames} {...attributes}>
      <path fill="#78288C" d="M24.8 47.7l22.9-22.9c.4-.4.4-1.1 0-1.6L24.8.3c-.4-.4-1.1-.4-1.6 0L.3 23.2c-.4.4-.4 1.1 0 1.6l22.9 22.9c.4.4 1.2.4 1.6 0z" ></path>
    </IconBase>
  );
};

SvgIcon.displayName = "IconDiamond";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg"};

export default SvgIcon;
/* eslint-enable */
