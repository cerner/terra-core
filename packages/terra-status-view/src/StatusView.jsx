import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import { FormattedMessage } from 'react-intl';
import Button from 'terra-button';
import Divider from 'terra-divider';
import styles from './StatusView.module.scss';

const cx = classNames.bind(styles);

const StatusViewVariants = {
  NODATA: 'no-data',
  NOMATCHINGRESULTS: 'no-matching-results',
  NOTAUTHORIZED: 'not-authorized',
  ERROR: 'error',
};

/* eslint-disable react/forbid-foreign-prop-types */
const propTypes = {
  /**
   * An array of objects containing terra-button properties. A key attribute is required for each object.
   * This array is used to render buttons in the bottom section.
   * Example:`[{ text: 'Button 1', key: 1, size: 'medium', variant: 'action', onClick: onClickFunction}]`
   */
  buttonAttrs: PropTypes.arrayOf(PropTypes.shape(Button.propTypes)),

  /**
   * Display a custom glyph. Overrides a variant's default glyph.
   */
  customGlyph: PropTypes.node,

  /**
   *  Aligns the component at the top of the container rather than "centered"
   */
  isAlignedTop: PropTypes.bool,

  /**
   * Determines if the glyph should be displayed.
   */
  isGlyphHidden: PropTypes.bool,

  /**
   * The descriptive text, displayed under the title.
   */
  message: PropTypes.string,

  /**
   * The title displayed under the glyph. Variants contain default titles that can be overriden by this prop.
   */
  title: PropTypes.string,

  /**
   * Sets the glyph and title using a pre-baked variant. One of the following: `no-data`,
   * `no-matching-results`, `not-authorized`, or `error`
   */
  variant: PropTypes.oneOf(['no-data', 'no-matching-results', 'not-authorized', 'error']),
};
/* eslint-enable react/forbid-foreign-prop-types */

const defaultProps = {
  buttonAttrs: [],
  customGlyph: undefined,
  isAlignedTop: false,
  isGlyphHidden: false,
  message: undefined,
  title: undefined,
  variant: undefined,
};

const generateButtons = (buttonAttrsArray) => {
  if (!buttonAttrsArray.length) {
    return undefined;
  }

  return buttonAttrsArray.map(button => <Button {...button} className={cx(['button', button.className])} />);
};

const StatusView = ({
  buttonAttrs,
  customGlyph,
  isAlignedTop,
  isGlyphHidden,
  message,
  title,
  variant,
  ...customProps
}) => {
  let glyphSection;
  if (customGlyph && !isGlyphHidden) {
    glyphSection = (
      <div className={cx('glyph')}>
        {customGlyph}
      </div>
    );
  } else if (variant && !isGlyphHidden) {
    glyphSection = (
      <div className={cx('glyph')}>
        <svg className={cx(variant)} />
      </div>
    );
  }

  let defaultTitle;
  if (variant) {
    defaultTitle = <FormattedMessage id={`Terra.status-view.${variant}`} />;
  }

  // Custom title takes precedence
  let titleSection;
  if (title || defaultTitle) {
    titleSection = (
      <p className={cx('title')}>
        {title || defaultTitle}
      </p>
    );
  }

  let messageSection;
  if (message) {
    messageSection = (
      <p className={cx('message')}>
        {message}
      </p>
    );
  }

  let actionSection;
  const buttons = generateButtons(buttonAttrs);
  if (buttons) {
    actionSection = (
      <div className={cx('actions')}>
        {buttons}
      </div>
    );
  }

  let dividerSection;
  if (titleSection && (messageSection || actionSection)) {
    dividerSection = (
      <div className={cx('divider')}>
        <Divider />
      </div>
    );
  }

  const outerViewClassNames = cx([
    'outer-view',
    { 'is-aligned-top': isAlignedTop },
    customProps.className,
  ]);

  const innerViewClassNames = cx([
    'inner-view',
  ]);

  return (
    <div {...customProps} className={outerViewClassNames}>
      <div className={cx('top-space')} />
      <div className={innerViewClassNames}>
        {glyphSection}
        {titleSection}
        {dividerSection}
        {messageSection}
        {actionSection}
      </div>
      <div className={cx('bottom-space')} />
    </div>
  );
};

StatusView.propTypes = propTypes;
StatusView.defaultProps = defaultProps;
export default StatusView;
export { StatusViewVariants };
