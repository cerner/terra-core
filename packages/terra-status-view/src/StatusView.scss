@import '~terra-mixins';

:local {
  .status-view {
    height: 100%;
    padding-bottom: var(--terra-status-view-padding-bottom, 20px);
    padding-top: var(--terra-status-view-padding-top, 20px);
    width: 100%;
  }

  .status-view-inner {
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 100%;
  }

  .glyph {
    flex: 0 0 auto;
    padding-bottom: 20px;
  }

  .glyph > svg {
    background-repeat: no-repeat;
    background-size: contain;
    height: 100px;
    width: 100px;
  }

  .glyph > .no-data {
    background: var(--terra-status-view-glyph-no-data, inline-svg("<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 48 48' class='is-bidi'><path d='M45.8 12c1.2 0 2.2 1 2.2 2.2v26.7c0 1.2-1 2.2-2.2 2.2H2.2C1 43 0 42 0 40.8V7.2C0 6 1 5 2.2 5H15c1.4.1 2.7.6 3.7 1.6l3.8 3.9c1 1 2.3 1.5 3.7 1.6l19.4-.1h.2z'/></svg>"));
  }

  .glyph > .no-matching-results {
    background: var(--terra-status-view-glyph-no-matching-results, inline-svg("<svg data-name='Layer 1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 48 48'><path d='M47.4 43.2l-13-13A19.66 19.66 0 0 0 38 19a19 19 0 1 0-7.9 15.4l13 13a2.05 2.05 0 0 0 2.9 0l1.4-1.4a1.93 1.93 0 0 0 0-2.8zM3 19a16 16 0 1 1 16 16A16 16 0 0 1 3 19z'/></svg>"));
  }

  .glyph > .not-authorized {
    background: var(--terra-status-view-glyph-not-authorized, inline-svg("<svg data-name='Layer 1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 48 48'><path d='M38.8 20H35v-9a11 11 0 0 0-22 0v9H9.2A2.22 2.22 0 0 0 7 22.2v23.7A2.2 2.2 0 0 0 9.2 48h29.7a2.22 2.22 0 0 0 2.2-2.2V22.2a2.38 2.38 0 0 0-2.3-2.2zM26 33.5V40h-4v-6.5a3.26 3.26 0 0 1-1-2.3 3 3 0 0 1 3-3 3.08 3.08 0 0 1 3 3 3.26 3.26 0 0 1-1 2.3zm6-14.9V20H16v-9a8 8 0 0 1 16 0z'/></svg>"));
  }

  .glyph > .error {
    background: var(--terra-status-view-glyph-error, inline-svg("<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 48 48'><path fill='#E50000' d='M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z'/><path fill='#FFF' d='M7 21.5h34v5H7v-5z'/></svg>"));
  }

  .glyph > .no-service {
    background: var(--terra-status-view-glyph-no-service, inline-svg("<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 48 48' class='is-bidi'><path d='M19 17.9c-4.1 0-7.8 1.7-10.5 4.3l-3.1-3.1c3.5-3.5 8.3-5.6 13.6-5.6s10.1 2.1 13.6 5.6L29.7 22c-.1 0-.1.1-.2.1-2.8-2.6-6.4-4.2-10.5-4.2zm15.7-.9l3.3-3.3C33.1 9 26.4 6 19 6S4.9 9 0 13.7L3.3 17c4-4 9.6-6.5 15.7-6.5S30.7 13 34.7 17zM21.2 28.9c-.7-.4-1.4-.6-2.2-.6-1.2 0-2.3.5-3.1 1.3l3.1 3.1 1.1-1.1c.3-.9.7-1.8 1.1-2.7zm1.8-2.4c1-1.1 2.2-2.1 3.5-2.9-2-1.6-4.6-2.6-7.4-2.6-3.2 0-6.2 1.3-8.4 3.4l3.1 3.1c1.3-1.3 3.2-2.2 5.2-2.2 1.5 0 2.8.4 4 1.2z'/><path d='M34 24c-6.6 0-12 5.4-12 12s5.4 12 12 12 12-5.4 12-12-5.4-12-12-12zm-9 12c0-5 4-9 9-9 1.9 0 3.7.6 5.2 1.7L26.7 41.2C25.6 39.7 25 37.9 25 36zm9 9c-1.9 0-3.7-.6-5.2-1.7l12.5-12.5c1 1.5 1.7 3.3 1.7 5.2 0 5-4 9-9 9z'/></svg>"));
  }

  .glyph > .custom {
    background: var(--terra-status-view-glyph-custom, none);
  }

  .message-content-group {
    align-items: center;
    display: block;
    flex-direction: column;
    padding-left: 10px;
    padding-right: 10px;
    width: 100%;
  }

  .title {
    color: var(--terra-status-view-title-color, #000);
    flex: 0 0 auto;
    flex-wrap: wrap;
    font-family: var(--terra-status-view-title-font-family, 'Roboto');
    font-size: var(--terra-status-view-title-font-size, 24px);
    text-align: center;
    width: 100%;
  }

  .message {
    color: var(--terra-status-view-message-color, #000);
    flex: 0 0 auto;
    flex-wrap: wrap;
    font-family: var(--terra-status-view-message-font-family, 'Roboto');
    font-size: var(--terra-status-view-message-font-size, 14px);
    padding-top: var(--terra-status-view-message-padding-top, 10px);
    text-align: var(--terra-status-view-message-text-align, center);
    width: 100%;
  }

  .divider {
    border-color: var(--terra-status-view-divider-color, #e8e9ea);
    display: var(--terra-status-view-divider-display, block);
    padding-top: var(--terra-status-view-divider-padding-top, 20px);
    width: 100%;
  }

  .buttons {
    display: flex;
    flex: 0 0 auto;
    flex-direction: var(--terra-status-view-buttons-flex-direction-small-width, row);
    justify-content: center;
    padding-left: 5px;
    padding-right: 5px;
    padding-top: var(--terra-status-view-buttons-padding-top, 10px);
    width: 100%;

    @media screen and (min-width: 768px) {
      flex-direction: row;
    }
  }

  .buttons > button {
    flex: var(--terra-status-view-button-flex, initial);
    margin-left: 5px;
    margin-right: 5px;
    margin-top: 10px;
  }
}
