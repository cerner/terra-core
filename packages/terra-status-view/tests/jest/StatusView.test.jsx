import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl } from 'terra-enzyme-intl';
import IconDiamond from 'terra-icon/lib/icon/IconDiamond';
import StatusView, { StatusViewVariants } from '../../src/StatusView';

// Catch console warnings when visuallyHiddenText prop has not been included for accessibility.
// This should be removed when visuallyHiddenText become required.
global.console = { log: jest.fn(), warn: jest.fn() };

const buttons = [
  {
    text: 'Action1',
    key: 1,
  }, {
    text: 'Action2',
    key: 2,
  },
];

// Snapshot Tests
it('should render a status view with custom glyph, title, message, and two buttons', () => {
  const statusView = (
    <StatusView customGlyph={<IconDiamond />} title="Test Title" message="Test Message" buttonAttrs={buttons} />
  );

  const wrapper = mountWithIntl(statusView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a no-data status view with a hidden glyph', () => {
  const statusView = (<StatusView variant={StatusViewVariants.NODATA} isGlyphHidden />);

  const wrapper = mountWithIntl(statusView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a no-matching-results status-view with top alignment', () => {
  const statusView = (<StatusView variant={StatusViewVariants.NOMATCHINGRESULTS} isAlignedTop />);

  const wrapper = mountWithIntl(statusView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a not-authorized status-view', () => {
  const statusView = (<StatusView variant={StatusViewVariants.NOTAUTHORIZED} />);

  const wrapper = mountWithIntl(statusView);
  expect(wrapper).toMatchSnapshot();
});

it('should render an error status-view', () => {
  const statusView = (<StatusView variant={StatusViewVariants.ERROR} />);

  const wrapper = mountWithIntl(statusView);
  expect(wrapper).toMatchSnapshot();
});
