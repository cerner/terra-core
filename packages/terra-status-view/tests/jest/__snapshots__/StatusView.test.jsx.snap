// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly applies the theme context className 1`] = `
<ThemeContextProvider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  theme={
    Object {
      "className": "orion-fusion-theme",
    }
  }
>
  <StatusView
    buttonAttrs={Array []}
    isAlignedTop={false}
    isGlyphHidden={false}
  >
    <div
      className="outer-view orion-fusion-theme"
    >
      <div
        className="top-space"
      />
      <div
        className="inner-view"
      />
      <div
        className="bottom-space"
      />
    </div>
  </StatusView>
</ThemeContextProvider>
`;

exports[`should render a no-data status view with a hidden glyph 1`] = `
<StatusView
  buttonAttrs={Array []}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isAlignedTop={false}
  isGlyphHidden={true}
  variant="no-data"
>
  <div
    className="outer-view"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div
      className="top-space"
    />
    <div
      className="inner-view"
    >
      <p
        className="title"
      >
        <FormattedMessage
          id="Terra.status-view.no-data"
          values={Object {}}
        >
          <span>
            Terra.status-view.no-data
          </span>
        </FormattedMessage>
      </p>
    </div>
    <div
      className="bottom-space"
    />
  </div>
</StatusView>
`;

exports[`should render a no-matching-results status-view with top alignment 1`] = `
<StatusView
  buttonAttrs={Array []}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isAlignedTop={true}
  isGlyphHidden={false}
  variant="no-matching-results"
>
  <div
    className="outer-view is-aligned-top"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div
      className="top-space"
    />
    <div
      className="inner-view"
    >
      <div
        className="glyph"
      >
        <div
          className="no-matching-results glyph-icon"
        />
      </div>
      <p
        className="title"
      >
        <FormattedMessage
          id="Terra.status-view.no-matching-results"
          values={Object {}}
        >
          <span>
            Terra.status-view.no-matching-results
          </span>
        </FormattedMessage>
      </p>
    </div>
    <div
      className="bottom-space"
    />
  </div>
</StatusView>
`;

exports[`should render a not-authorized status-view 1`] = `
<StatusView
  buttonAttrs={Array []}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isAlignedTop={false}
  isGlyphHidden={false}
  variant="not-authorized"
>
  <div
    className="outer-view"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div
      className="top-space"
    />
    <div
      className="inner-view"
    >
      <div
        className="glyph"
      >
        <div
          className="not-authorized glyph-icon"
        />
      </div>
      <p
        className="title"
      >
        <FormattedMessage
          id="Terra.status-view.not-authorized"
          values={Object {}}
        >
          <span>
            Terra.status-view.not-authorized
          </span>
        </FormattedMessage>
      </p>
    </div>
    <div
      className="bottom-space"
    />
  </div>
</StatusView>
`;

exports[`should render a status view with custom glyph, title, message, and two buttons 1`] = `
<StatusView
  buttonAttrs={
    Array [
      Object {
        "key": 1,
        "text": "Action1",
      },
      Object {
        "key": 2,
        "text": "Action2",
      },
    ]
  }
  customGlyph={
    <IconDiamond
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isAlignedTop={false}
  isGlyphHidden={false}
  message="Test Message"
  title="Test Title"
>
  <div
    className="outer-view"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div
      className="top-space"
    />
    <div
      className="inner-view"
    >
      <div
        className="glyph"
      >
        <IconDiamond
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <IconBase
            ariaLabel={null}
            className="IconDiamond"
            focusable="false"
            height="1em"
            isBidi={false}
            isSpin={false}
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              aria-hidden="true"
              className="tui-Icon icon IconDiamond"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24.8 47.7l22.9-22.9c.4-.4.4-1.1 0-1.6L24.8.3c-.4-.4-1.1-.4-1.6 0L.3 23.2c-.4.4-.4 1.1 0 1.6l22.9 22.9c.4.4 1.2.4 1.6 0z"
                fill="#78288C"
              />
            </svg>
          </IconBase>
        </IconDiamond>
      </div>
      <p
        className="title"
      >
        Test Title
      </p>
      <div
        className="divider"
      >
        <Divider>
          <hr
            aria-hidden="true"
            className="divider"
          />
        </Divider>
      </div>
      <p
        className="message"
      >
        Test Message
      </p>
      <div
        className="actions"
      >
        <Button
          className="button"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          key="1"
          text="Action1"
          type="button"
          variant="neutral"
        >
          <button
            aria-disabled={false}
            className="button neutral button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            type="button"
          >
            <span
              className="button-label text-only"
            >
              <span
                className=""
              >
                Action1
              </span>
            </span>
          </button>
        </Button>
        <Button
          className="button"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          key="2"
          text="Action2"
          type="button"
          variant="neutral"
        >
          <button
            aria-disabled={false}
            className="button neutral button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            type="button"
          >
            <span
              className="button-label text-only"
            >
              <span
                className=""
              >
                Action2
              </span>
            </span>
          </button>
        </Button>
      </div>
    </div>
    <div
      className="bottom-space"
    />
  </div>
</StatusView>
`;

exports[`should render an error status-view 1`] = `
<StatusView
  buttonAttrs={Array []}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isAlignedTop={false}
  isGlyphHidden={false}
  variant="error"
>
  <div
    className="outer-view"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div
      className="top-space"
    />
    <div
      className="inner-view"
    >
      <div
        className="glyph"
      >
        <div
          className="error glyph-icon"
        />
      </div>
      <p
        className="title"
      >
        <FormattedMessage
          id="Terra.status-view.error"
          values={Object {}}
        >
          <span>
            Terra.status-view.error
          </span>
        </FormattedMessage>
      </p>
    </div>
    <div
      className="bottom-space"
    />
  </div>
</StatusView>
`;
